src/analysis/Environments/Environments.ts

```diff
@@ -384,7 +391,7 @@ export const Environments = (props: EnvironmentsProps): ReactNode => {
     // @ts-expect-error
     [sort.direction],
     runtimes
-  );
+  ) as unknown as typeof runtimes;
```

Comment:
yes, because lodash/fp is horrible with type flow...

Reply:
My experience has been that an issue with the output's type is often caused by an issue with the input's type.

Reply:
agreed, I saw that input can be undefined, but that wasn't the only issue in some cases like here.... at any rate, I'd love to rip this data conversion stuff out to a helper where it can be more focused.

Reply:
I'd like to rework all this data transformation to be a more focused helper to both remove bloat from this component, and to make it less wonky on the types.